src/hotspot/os/posix/os_posix.cpp

```diff
@@ -947,27 +947,32 @@ void os::naked_yield() {
 char* os::build_agent_function_name(const char *sym_name, const char *lib_name,
                                     bool is_absolute_path) {
   char *agent_entry_name;
-  size_t len;
-  size_t name_len;
+  size_t len = 0;
+  size_t name_len = 0;
   size_t prefix_len = strlen(JNI_LIB_PREFIX);
   size_t suffix_len = strlen(JNI_LIB_SUFFIX);
   const char *start;
+  bool need_underscore = false;
 
   if (lib_name != nullptr) {
-    name_len = strlen(lib_name);
     if (is_absolute_path) {
       // Need to strip path, prefix and suffix
       if ((start = strrchr(lib_name, *os::file_separator())) != nullptr) {
         lib_name = ++start;
       }
-      if (strlen(lib_name) <= (prefix_len + suffix_len)) {
+      name_len = strlen(lib_name);
+      if (name_len <= (prefix_len + suffix_len)) {
         return nullptr;
       }
       lib_name += prefix_len;
       name_len = strlen(lib_name) - suffix_len;
+    } else {
+      name_len = strlen(lib_name);
     }
+    need_underscore = true;
   }
-  len = (lib_name != nullptr ? name_len : 0) + strlen(sym_name) + 2;
+   // Total buffer length to allocate - includes null terminator.
+  len = strlen(sym_name) + (need_underscore ? 1 : 0) + name_len + 1;
   agent_entry_name = NEW_C_HEAP_ARRAY_RETURN_NULL(char, len, mtThread);
```

Comment:
Preexisting: If we fail to allocate memory, we return NULL, which results in various places in confusing "cannot find agent function" messages. Maybe better would be a clear native oom exit here?

Reply:
That kind of change in functionality is not a part of this PR.  Plus it is a moot point as we would never get a native OOM during VM init in any realistic circumstances.

Reply:
Sorry, you are right - I should have not responded this late at night :) If we are going to abort the VM we should at least report the correct reason.

Reply:
Nope I retract that again. This code is used for both command-line agents and dynamically attaching agents, so this may not happen during init, so we do not want to abort on OOM.